src/chapters/setup-intro-data-types/more-data-types.rst

```diff
@@ -2,258 +2,321 @@ More Data Types
+instead of square brackets ``[ ]``.
 
-To access array elements, we use square brackets, as with Python lists.
+To access array elements, we use square brackets and *zero-based indexing*.
 
-.. code:: java
+.. sourcecode:: java
 
-   int anInt = someInts[0];
+   int anInt = someOtherInts[4];
+   // anInt stores the integer 5.
 
-As with Python lists, arrays have *zero-based indexing*.
+Arrays in Java may NOT change size once created. This is limiting and not very
+practical. Thankfully, Java provides more flexible ways to store data, which we
+will explore in a later lesson. These objects will allow us to rearrange, add
+to, or remove data.
 
-Aside from using arrays to build some simple loop examples in the next
-lesson, we’ll only use them in special cases. However, they’re a core
-part of Java, so it’s good to know how they work.
+Aside from using arrays to build some simple loop examples, we will only use
+them in special cases. However, they are a core part of Java, so it’s good to
+know how they work.
 
 Static Methods
---------------
+---------------
 
-In pure object-oriented languages like Java and C#, we don’t have
-functions in the sense you’re used to. For example, functions may not be
-declared outside of a class. Within the context of a class, functions
-are referred to as **methods**. We will adopt this terminology from now
-on, and you are encouraged not to refer to methods as “functions” when
-you are talking about Java code.
+If you are familiar with another programming language, then you most likely
+created chunks of re-usable code called *functions*. These blocks of code make
+writing programs more efficient and flexible. Java embraces the same concept,
+but it approaches the idea in a very specific way.
 
-We’ll dive into learning about classes and objects in Java soon enough,
-but until we do, we’ll frequently need to write methods, so we should
-understand a little bit about them. In particular, we’ll use **static
-methods**, which behave similarly to functions as you knew them in
-Python.
+In Java, functions may NOT be declared outside of a class. Even a simple
+function that checks if an integer is even needs to be defined within a class.
 
-A static method is one with the ``static`` keyword, as our ``main``
-method above has:
+.. index:: ! methods
 
-.. code:: java
+Within the context of a class, functions are referred to as **methods**, and we
+will adopt this terminology from now on.
 
-   public static void main(String[] args)
-   {
-       // some code
-   }
+.. admonition:: Warning
 
-We’ve already explored each element of this line, however, we haven’t
-really shown you how you might create your own methods in other
-contexts. To do so involves using a different name for our method, and
-adjusting the return type and parameter types accordingly.
+   Be prepared for push-back and negative judgement from Java coders if you
+   accidentally refer to methods as “functions”.
```

Comment:
i agree its good to stress they should call these methods but phrasing in this way opens the door to negative feelings! we should all be prepared for negative judgement from other programmers but thats not enough of a reason to change a behavior, imo!

Reply:
Just another dose of optional snark.  I re-wrote the warning.  Let me know what you think.